crates/nu-cmd-extra/src/extra/bits/mod.rs

```diff
@@ -95,7 +100,68 @@ fn get_input_num_type(val: i64, signed: bool, number_size: NumberBytes) -> Input
                     InputNumType::Eight
                 }
             }
-            NumberBytes::Invalid => InputNumType::Four,
         }
     }
 }
+
+fn binary_op<F>(lhs: &Value, rhs: &Value, little_endian: bool, f: F) -> Value
+where
+    F: Fn((i64, i64)) -> i64,
+{
+    let span = lhs.span();
+    match (lhs, rhs) {
+        (Value::Int { val: lhs, .. }, Value::Int { val: rhs, .. }) => {
+            Value::int(f((*lhs, *rhs)), span)
+        }
+        (Value::Binary { val: lhs, .. }, Value::Binary { val: rhs, .. }) => {
+            let (lhs, rhs, max_len, min_len) = match (lhs.len(), rhs.len()) {
+                (max, min) if max > min => (lhs, rhs, max, min),
+                (min, max) => (rhs, lhs, min, max),
```

Comment:
I'm getting a crash when the left operand is smaller:

![image](https://github.com/nushell/nushell/assets/45539777/db79a77b-1c82-47b8-afb6-eeb6a7a10d3e)

I think this should fix it? Make sure to add a test/example for that scenario first, though (I think an example might be better, so users have a better idea of `and`'s behavior)
```suggestion
                (min, max) => (rhs, lhs, max, min),
```
Also, sorry I didn't review this PR thoroughly earlier, been busy

Reply:
right thanks, that was what I intended and not sure how I missed it
I am currently compiling the fix to verify it is correct, then I will add a few more tests as well